src/universal/components/FileInput/FileInput.js

```diff
@@ -0,0 +1,31 @@
+import React, {Component, PropTypes} from 'react';
+
+class FileInput extends Component {
+  static propTypes = {
+    input: PropTypes.object,
+    previousValue: PropTypes.string
+  };
+
+  constructor(props) {
+    super(props);
+    this.onChange = this.onChange.bind(this);
```

Comment:
this is fine, there's nothing wrong with binding in the constructor, but the rest of the components do the pattern like:

`onChange = (e) => {}`
by thunkifying the method, it achieves the same thing.